### PR TITLE
NOISSUE: raise error if ClusterDeployment status changed to error

### PR DESCRIPTION
**Pull request by RazRegev**
Instead of waiting 2 hours to find out the installation has failed,
raise error immediately if the status changes to error.

### CODE REVIEW

File: discovery-infra/test_infra/helper_classes/kube_helpers/cluster_deployment.py [hunk 1]
```diff
@@ -343,20 +355,16 @@ def _wait_for_sufficient_agents_number() -> List[Agent]:
...
-            sleep_seconds=1,
-            timeout_seconds=timeout,
-            waiting_for=f'cluster {self.ref} to be installed',
-        )
+    def wait_to_be_installed(
+            self, 
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_INSTALLATION_COMPLETE_TIMEOUT,
+    ) -> None:
+        self.wait_for_state('installed', timeout, raise_on_states=FAILURE_STATES)
+        assert self.get()['spec'].get('installed') is True
```
Comment (osherdp) [inline discovery-infra/test_infra/helper_classes/kube_helpers/cluster_deployment.py:363]: maybe remove ``is True``?

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
/hold
added one optional comment

**Review by nmagnezi — APPROVED** [review]

**Pull request merged**